lib/webengine_kiosk.ex

```diff
@@ -59,19 +52,36 @@ defmodule WebengineKiosk do
   * `width: pixels` - when not fullscreen, the window is this width
   * `height: pixels` - when not fullscreen, the window is this height
   """
+
   @spec start_link(Keyword.t(), GenServer.options()) :: {:ok, pid} | {:error, term}
   def start_link(args, genserver_opts \\ []) do
-    with :ok <- Options.check_args(args) do
-      GenServer.start_link(__MODULE__, args, genserver_opts)
+    # subscriptions are stored in an agent in order to retain them during restarts.
+    {:ok, agent} = Agent.start_link(fn -> [] end)
+
+    children = [
+      {WebengineKiosk.Kiosk, {%{agent: agent, args: args, supervised: true}, genserver_opts}}
+    ]
+
+    opts = [strategy: :rest_for_one]
+    Supervisor.start_link(children, opts)
+  end
+
+  @doc """
+  Start single instance of kiosk
+  """
+  @spec start(Keyword.t(), GenServer.options()) :: {:ok, pid} | {:error, term}
+  def start(args, genserver_opts \\ []) do
+    with :ok <- WebengineKiosk.Options.check_args(args) do
+      GenServer.start_link(WebengineKiosk.Kiosk, args, genserver_opts)
```

Comment:
I'm not sure i agree with having the function start here. It's actually creating a link and its name is misleading. I think it should be removed

Reply:
Ok, i will refactor this